### PR TITLE
SavePlot1D hide spectra

### PR DESCRIPTION
**Pull request by peterfpeterson**
Add the ability to specify which spectra to show. This will allow NOMAD's autoreduction to use the standard plotting rather than have one hand-written in their script.

**To test:**

Load a dataset and try to export the plot with and without `SpectraList` specified for both images (e.g. `.jpeg`) and plotly.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SavePlot1D.py [hunk 1]
```diff
@@ -104,6 +108,15 @@ def getData(self, ws, wkspIndex, label=''):
 
         return (x, y, label)
 
+    def showSpectrum(self, ws, wkspIndex):
+        spectraNum = ws.getSpectrum(wkspIndex).getSpectrumNo()
+
+        # user not specifying which spectra means show them all
+        if len(self.visibleSpectra) <= 0:
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SavePlot1D.py]: Can an array length be negative?

Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SavePlot1D.py]: No, but the return value of `len` is a signed integer so I'd rather be paranoid.

**Comment by martyngigg** [comment]
@peterfpeterson There's a merge conflict

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SavePlot1D.py (hunk 1 above)
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SavePlot1D.py]: http://stackoverflow.com/questions/15650878/returning-from-len-when-64-bits

**Comment by AndreiSavici** [comment]
:shipit:

**Pull request merged**